src/logic/calculate.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-param-reassign, eqeqeq, no-undef */
+import operate from './operate';
+
+const calculate = (buttonName, dataObj) => {
+  const val = parseFloat(buttonName);
+  if (val.isNaN()) {
+    dataObj.total += val;
+  } else if (buttonName == '+/-') {
+    dataObj.total *= -1;
+    dataObj.next *= -1;
+  } else if (buttonName == 'AC') {
+    dataObj.total = '';
+    dataObj.next = '';
+    dataObj.operation = '';
+  } else if (buttonName == '=') {
```

Comment:
I think you should use `===`.
Example; `} else if (buttonName === '+/-') {`
we need to check the dataType and compares the two values.